From 992fd7c77d75c6267b615a3e9e982349639b9e51 Mon Sep 17 00:00:00 2001
From: Anton Arnold <aarnold.kde@antonarnold.de>
Date: Wed, 11 Dec 2019 23:37:59 +0200
Subject: Fix color-options layout & KSaneOptCombo::setValue()

Summary:
BUG: 414601
BUG: 414602

This diff contains bugfixes that I fixed while developing the scan area+option ordering enhancements but are not related directly to them.

The items in the color layout where not updated correctly.
Remove the group separator before trying to write a value to a combo-box option.

Reviewers: sars, #kde_applications

Reviewed By: sars

Subscribers: aacid

Differential Revision: https://phabricator.kde.org/D25588
---
 src/ksanewidget_p.cpp         | 6 +++---
 src/options/ksaneoptcombo.cpp | 5 +++--
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/ksanewidget_p.cpp b/src/ksanewidget_p.cpp
index 5a4d00b..9646fdb 100644
--- a/src/ksanewidget_p.cpp
+++ b/src/ksanewidget_p.cpp
@@ -429,9 +429,9 @@ void KSaneWidgetPrivate::createOptInterface()
         }
     }
     // Color Options
-    for (int i = 0; i < basic_layout->count(); ++i) {
-        if (basic_layout->itemAt(i) && basic_layout->itemAt(i)->widget()) {
-            tmpOption = qobject_cast<KSaneOptionWidget *>(basic_layout->itemAt(i)->widget());
+    for (int i = 0; i < color_lay->count(); ++i) {
+        if (color_lay->itemAt(i) && color_lay->itemAt(i)->widget()) {
+            tmpOption = qobject_cast<KSaneOptionWidget *>(color_lay->itemAt(i)->widget());
             if (tmpOption) {
                 labelWidth = qMax(labelWidth, tmpOption->labelWidthHint());
             }
diff --git a/src/options/ksaneoptcombo.cpp b/src/options/ksaneoptcombo.cpp
index 7156b1a..28263a2 100644
--- a/src/options/ksaneoptcombo.cpp
+++ b/src/options/ksaneoptcombo.cpp
@@ -34,6 +34,7 @@
 
 #include <QDebug>
 #include <QIcon>
+#include <QLocale>
 
 namespace KSaneIface
 {
@@ -341,7 +342,7 @@ bool KSaneOptCombo::setValue(const QString &val)
     case SANE_TYPE_INT:
         tmp = val.left(val.indexOf(QLatin1Char(' '))); // strip the unit
         // accept float formatting of the string
-        i = (int)(tmp.toFloat(&ok));
+        i = (int)(QLocale::system().toFloat(tmp,&ok));
         if (ok == false) {
             return false;
         }
@@ -350,7 +351,7 @@ bool KSaneOptCombo::setValue(const QString &val)
         break;
     case SANE_TYPE_FIXED:
         tmp = val.left(val.indexOf(QLatin1Char(' '))); // strip the unit
-        f = tmp.toFloat(&ok);
+        f = QLocale::system().toFloat(tmp,&ok);
         if (ok == false) {
             return false;
         }
-- 
cgit v1.1

